File: packages/shared-types/src/report.ts
/** Mirrors `app/schemas/report.py`. */

export interface ReportCreate {
  format: "pdf";
}

export interface ReportResponse {
  id: string;
  analysis_id: string;
  format: string;
  status: "completed" | "failed" | "expired";
  created_at: string;
  size_bytes: number | null;
  sha256: string | null;
  page_count: number | null;
  summary: Record<string, unknown>;
  error_code: string | null;
  error_message: string | null;
}

export interface ReportListResponse {
  items: ReportResponse[];
}

/** Short-lived signed URL for the rendered file; never a storage key. */
export interface ReportFileLink {
  url: string;
  expires_in_seconds: number;
  content_type: string;
}

File: services/api/app/schemas/report.py
import uuid
from datetime import datetime
from typing import Any, Literal

from pydantic import BaseModel


class ReportCreate(BaseModel):
    format: Literal["pdf"] = "pdf"


class ReportResponse(BaseModel):
    id: uuid.UUID
    analysis_id: uuid.UUID
    format: str
    status: Literal["completed", "failed", "expired"]
    created_at: datetime
    size_bytes: int | None
    sha256: str | None
    page_count: int | None
    summary: dict[str, Any]
    error_code: str | None
    error_message: str | None


class ReportListResponse(BaseModel):
    items: list[ReportResponse]


class ReportFileLink(BaseModel):
    """Short-lived signed URL for the rendered file; never a storage key."""

    url: str
    expires_in_seconds: int
    content_type: str
